source/hardware/controllers/native_instruments_traktor_kontrol_s3.rst

```diff
@@ -0,0 +1,167 @@
+
+  - Absolute Mode:
+
+     - Keylock: Press to toggle keylock mode.
+     - Sync: Press to beatsync, or press and hold to activate Sync Lock Mode.
+     - Pitch slider: Adjusts playback speed.
+     - Shift + Pitch slider: Adjusts musical key
+  - Relative Mode:
+
+     - Keylock: Press to toggle keylock mode.
+     - Sync: Press to beatsync, or press and hold to activate Sync Lock Mode.
+     - Pitch slider: Adjusts playback speed.
+     - Keylock + Pitch Slider: adjusts musical key
+     - Shift + Pitch Slider: Allows the user to move the slider without any effect.
+
+Effects
+~~~~~~~
+
+The FX setup is unusual on this controller.  Each deck has a single toggle button for effects, and on the right-hand side of the mixer there are five buttons that determine which effects are applied to every channel that has effects on.  This means it is not possible to use the controller to select one effect for one deck, and another effect for another.  You can still make these choices in the Mixxx UI, however.
+
+When the Filter Enable button is off, the knob still controls the default QuickEffect, even if that button is not lit in the FX section. When the Filter Enable button is on, the QuickEffect is only enabled if the Filter FX Select button is on.  This means the QuickEffect is available on channels that don't have any other effects active.
+
+When turning Filter Enable off, Mixxx will use soft takeover so the QuickEffect doesn't suddenly activate.
+
+Mapping options
+~~~~~~~~~~~~~~~
+
+There are two user-friendly customizations possible on the S3:
+
+  1. Toggle between Absolute and Relative pitch slider mode.
+  2. Customize the colors for decks A, B, C, and D.
+
+Making these changes is still a little awkward and we will be making
+controller preferences easier to change in the future. For now you'll
+have to make a small change to the mapping script file. Don't worry, the
+actual edit only involves replacing a single word in a text file.
+
+1.  Open Mixxx Preferences and select the Kontrol S3 in the side list.
+2.  There will be a box labeled Preset Info, and that box will have a section
+    labeled "Script Files".
+3.  Select "Traktor-Kontrol-S3-hid-scripts.js".
+4.  Either the file should open in an editor, or you should see a file
+    browser window with that file selected. If you see a file browser,
+    right click the file and select an option to edit it.
+5.  At the top of the file will be short instructions explaining how to edit
+    the file.
+
+Changes you make will take effect as soon as you save the file.
+
+References
+----------
+
+  - `Manufacturer's product
+    page <https://www.native-instruments.com/en/products/traktor/dj-controllers/traktor-kontrol-s3/>`__
```

Comment:
Currently, we have links as part of the first section, directly below the text (I. E. Not in a separate subsection).

We can discuss if putting them into a different "References" section at the end is better, but then we should do that for *all* mappings consistently, not just this one.

Reply:
let me know if the new placement is ok